rotkehlchen/exchanges/coinbase.py

```diff
@@ -702,11 +715,118 @@ def query_online_deposits_withdrawals(
+                asset = asset_from_coinbase(raw_data['amount']['currency'], time=timestamp)
+                native_amount = deserialize_asset_amount(raw_data['native_amount']['amount'])
+                native_asset = asset_from_coinbase(raw_data['native_amount']['currency'])
+                rate = ZERO
+                if amount and native_amount != ZERO:
+                    rate = native_amount / amount
+                if 'details' in raw_data:
+                    notes = (f"{raw_data['details']['title']} "
+                             f"{raw_data['details']['subtitle']} "
+                             f"{raw_data['details']['header']}")
+                else:
+                    notes = ''
+                return LedgerAction(
+                    identifier=0,
+                    location=Location.COINBASE,
+                    action_type=action_type,
+                    timestamp=timestamp,
+                    asset=asset,
+                    amount=amount,
+                    rate=Price(rate),
+                    rate_asset=native_asset,
+                    link=str(raw_data['id']),
+                    notes=notes,
+                )
+        except UnknownAsset as e:
+            self.msg_aggregator.add_warning(
+                f'Found coinbase transaction with unknown asset '
+                f'{e.asset_name}. Ignoring it.',
+            )
+        except UnsupportedAsset as e:
+            self.msg_aggregator.add_warning(
+                f'Found coinbase transaction with unsupported asset '
+                f'{e.asset_name}. Ignoring it.',
+            )
+        except (DeserializationError, KeyError) as e:
+            msg = str(e)
+            if isinstance(e, KeyError):
+                msg = f'Missing key entry for {msg}.'
+            self.msg_aggregator.add_error(
+                'Unexpected data encountered during deserialization of a coinbase '
+                'asset movement. Check logs for details and open a bug report.',
+            )
+            log.error(
+                f'Unexpected data encountered during deserialization of coinbase '
+                f'asset_movement {raw_data}. Error was: {str(e)}',
+            )
+        return None
+
+    def query_exchange_specific_history(
+            self,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> List[LedgerAction]:
+        account_data = self._api_query('accounts')
+        account_ids = self._get_account_ids(account_data)
+        raw_data = []
+        for account_id in account_ids:
+            txs = self._api_query(f'accounts/{account_id}/transactions')
+            for tx in txs:
+                if 'type' not in tx:
+                    continue
+                if tx['type'] in ['send', 'inflation_reward'] and 'from' in tx:
+                    if 'resource' in tx['from'] and tx['from']['resource'] == 'user':
+                        raw_data.append(tx)
```

Comment:
Why use two ifs? Just use one big if.

```suggestion
                if tx['type'] in ('send', 'inflation_reward') and 'from' in tx and 'resource' in tx['from'] and tx['from']['resource'] == 'user':
                        raw_data.append(tx)
```